packages/cli/src/oclif/commands/promote.js

```diff
@@ -0,0 +1,153 @@
+const _ = require('lodash');
+const colors = require('colors/safe');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const utils = require('../../utils');
+const { buildFlags } = require('../buildFlags');
+
+const serializeErrors = errors => {
+  const opener = 'Promotion failed for the following reasons:\n\n';
+  if (typeof errors[0] === 'string') {
+    // errors is an array of strings
+    return opener + errors.map(e => `* ${e}`).join('\n');
+  }
+
+  const issues = utils.flattenCheckResult({ errors: errors });
+  return (
+    opener +
+    issues
+      .map(i => `* ${i.method}: ${i.description}\n ${colors.gray(i.link)}`)
+      .join('\n')
+  );
+};
+
+class PromoteCommand extends BaseCommand {
+  async perform() {
+    await utils.checkCredentials();
+
+    const version = this.args.version;
+
+    let shouldContinue;
+    const changelog = await utils.getVersionChangelog(version);
+    if (changelog) {
+      this.log(colors.green(`Changelog found for ${version}`));
+      this.log(`\n---\n${changelog}\n---\n`);
+
+      shouldContinue = await this.confirm(
+        'Would you like to continue promoting with this changelog?'
+      );
+    } else {
+      this.log(
+        `${colors.yellow(
+          'Warning!'
+        )} Changelog not found. Please create a CHANGELOG.md file in a format similar to ${colors.cyan(
+          'https://gist.github.com/xavdid/b9ede3565f1188ce339292acc29612b2'
+        )} with user-facing descriptions.`
+      );
+
+      shouldContinue = await this.confirm(
+        'Would you like to continue promoting without a changelog?'
+      );
+    }
+
+    if (!shouldContinue) {
+      throw new Error('Cancelled promote.');
+    }
+
+    const app = await utils.getLinkedApp();
+    this.log(
+      `Preparing to promote version ${version} of your app "${app.title}".`
+    );
+
+    const body = {};
+    if (changelog) {
+      body.changelog = changelog;
+    }
+
+    this.startSpinner(`Verifying and promoting ${version}`);
+
+    const url = `/apps/${app.id}/versions/${version}/promote/production`;
+    try {
+      await utils.callAPI(
+        url,
+        {
+          method: 'PUT',
+          body
+        },
+        true
+      );
+    } catch (response) {
+      this.stopSpinner();
+
+      const activationUrl = _.get(response, 'json.activationInfo.url');
```

Comment:
super minor, but I've seen a linting rule that enforces the second argument to `_.get` be an array of strings (`['json', 'activationInfo', 'url']`), and I like that. For longer chains, prettier will put each item on a new line, which makes it nice and readable.

Separately, if it's easy, I'd destructure the `lodash` require (`const { get } = require('lodash')`) to match the other more explicit requiring.

Reply:
Changed it to array path in 93815f0.

I'd keep `_.get` though. I like explicit requiring most of the time, but this one is an exception. `get` is a super generic name. Keeping it `_.get` actually makes it clear that it's lodash's `get` without requiring us to look for where we import it in the top section.

Reply:
Fine by me! the other (maybe silly) option is to rename it (`const {get: lodashGet} = require('lodash')`), but I don't care that much. 

Worth nothing that [optional chaining](https://github.com/tc39/proposal-optional-chaining) is in stage 3, which means it'll be in JS for real in the next year (and typescript starting in nov 👀), so we'll eventually be able to do `response?.json?.activationInfo?.url`! That has no bearing on this pr, but it's fun info.